gomock/matchers.go

```diff
@@ -139,3 +190,52 @@ func Not(x interface{}) Matcher {
 func AssignableToTypeOf(x interface{}) Matcher {
 	return assignableToTypeOfMatcher{reflect.TypeOf(x)}
 }
+
+// PredFunc is a Matcher that matches if the one argument function parameter
+// returns true for the given input
+//
+// Example usage:
+//
+// 		mock.EXPECT().
+// 			Handle(gomock.PredFunc(func(x interface{}) bool { return x.(int) > 5 })).
+// 			Return(true)
+//
+func PredFunc(f func(interface{}) bool, name string) Matcher {
+	return predFuncMatcher{predFunc: f, name: name}
+}
+
+// HasPrefix is a Matcher that matches if the input has the arg as a prefix
+//
+// Example usage:
+//
+// 		mock.EXPECT().
+// 			Handle(gomock.HasPrefix("abc")).
+// 			Return(true)
+//
+func HasPrefix(prefix string) Matcher {
+	return hasPrefixMatcher{prefix: prefix}
+}
+
+// HasSuffix is a Matcher that matches if the input has the arg as a suffix
+//
+// Example usage:
+//
+// 		mock.EXPECT().
+// 			Handle(gomock.HasSuffix(".exe")).
+// 			Return(true)
+//
+func HasSuffix(suffix string) Matcher {
+	return hasSuffixMatcher{suffix: suffix}
+}
+
+// Regexp is a Matcher that matches if the input matches the regular expression string arg
+//
+// Example usage:
+//
+// 		mock.EXPECT().
+// 			Handle(gomock.Regexp(".exe$")).
```

Comment:
super nit: Can you throw a backslash (`\`) in front of the period (`.`)? I'm assuming you are demonstrating a test that looks for a windows executable extension (`.exe`), but the regexp would actually match against anything ending in `exe`.

I feel silly for bringing it up, but all the same.